mobile/ios/ios-device.ts

```diff
@@ -322,11 +328,19 @@ export class IOSDevice implements Mobile.IIOSDevice {
 		}).future<void>()();
 	}
 
-    public debug(packageFile: string, packageName: string): IFuture<void> {
-        return (() => {
-            this.$errors.fail({formatStr:"this will come in a future version", suppressCommandHelp: true});
-        }).future<void>()();
-    }
+	public debug(packageFile: string, packageName: string): IFuture<void> {
+		return (() => {
+
+			var debugging: IOSDeviceDebugging = this.$injector.resolve(IOSDeviceDebugging, {
```

Comment:
While using manual injector resolve is correct, it makes writing unit tests harder, because one have to perform lifetime management of the injector itself. Consider using constructor injection if possible.